### DISCUSSION TITLE
How far can it go?

### DISCUSSION THREAD

**Discussion by cocathail** [imohitmayank/jaal#48] [Q&A]
So I've been thinking about implementing Jaal for new project. My question is, does anyone have any indication how performant it is? How many nodes and edges can you render before it breaks?

**Comment by imohitmayank** [comment]
Hey @cocathail, sorry for the late reply. So while I have not performed an extensive analysis on the performance, I have tried it on hundreds of nodes and a couple of hundreds of edges and it works great.

**Reply by nikostro** [reply]
Hey @imohitmayank! Do you have an idea of what might hold Jaal back, performance-wise? I'm interested in plotting <10k nodes with ~100k edges, do you think it would manage if left to run long enough?

**Reply by imohitmayank** [reply]
Hey @nikostro, most of the time is consumed by the physics engine that organizes the nodes and edges for "better" visualization.   While I have never tried with that much data, there is a hack to reduce the number of iterations performed by the physics engine. For your data size, my suggestion will be to keep the iterations in single digit ([link](https://github.com/imohitmayank/jaal#using-visjs-settings)). Do note, this might lead to a messy network.
